src/static/js/templates/generate_msamd_dropdown.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+from argparse import ArgumentParser
+from re import match
+from urllib2 import urlopen
+from csv import DictReader
+from shutil import copy
+from sys import exit
+from datetime import datetime
+
+DEFAULT_URL = 'https://github.com/cfpb/api/raw/master/resources/datasets/hmda/code_sheets/msamd.csv'
+
+# Parse the command line arguments
+parser = ArgumentParser()
+group = parser.add_mutually_exclusive_group()
+group.add_argument("-u", "--url", help="URL to use as the CSV data source")
+group.add_argument("-f", "--file", help="File to use as the CSV data source")
+parser.add_argument("-o", "--out", help="HTML file to backup and update")
+args = parser.parse_args()
+
+csv_feed = None
+if args.url:
+    # If the url is a github file, and is an html link rather than a raw link,
+    # update the url to refer to the raw file
+    github_regex = '.*github.*/blob/.*'
+    if match(github_regex, args.url):
+        csv_url = str.replace(args.url, '/blob/', '/raw/', 1)
+    else:
+        csv_url = args.url
+    csv_feed = urlopen(csv_url)
+elif args.file:
+    csv_feed = open(args.file)
+else:
+    csv_feed = urlopen(DEFAULT_URL)
+
+
+def rangeify(years_list):
```

Comment:
I didn't dig in, but an unsorted list gives me a different answer from a sorted one. (This should probably get a unit test.)